File: Selenium/Activities/Activity5.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Open the link in browser
    driver.get("https://www.training-support.net/selenium/input-events")

    # Print page title
    print("Page Title is: " + driver.title)

    obj = driver.find_element(By.CLASS_NAME, "active")

    actions = webdriver.ActionChains(driver)

    # Left click
    actions.click(obj).pause(20).perform()
    obj = driver.find_element(By.CLASS_NAME, "active")
    print("Text: " + obj.text)

    # Left click
    actions.double_click(obj).perform()
    obj = driver.find_element(By.CLASS_NAME, "active")
    print("Text: " + obj.text)

    # Left click
    actions.context_click(obj).perform()
    obj = driver.find_element(By.CLASS_NAME, "active")
    print("Text: " + obj.text)

    driver.quit()

File: Selenium/Activities/Activity5.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity5 {

    public static void main(String[] args) {
        // Installing Firefox driver
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();

        // Open a page
        driver.get("https://www.training-support.net/selenium/input-events");

        // Print page title
        System.out.println("Page Title is: " + driver.getTitle());

        WebElement obj = driver.findElement(By.className("active"));

        Actions builder = new Actions(driver);

        // Left click
        builder.click(obj).pause(1000).build().perform();
        obj = driver.findElement(By.className("active"));
        System.out.println("Text: " + obj.getText());

        // Double click
        builder.doubleClick(obj).pause(1000).build().perform();
        obj = driver.findElement(By.className("active"));
        System.out.println("Text: " + obj.getText());

        // Right click
        builder.contextClick(obj).pause(1000).build().perform();
        obj = driver.findElement(By.className("active"));
        System.out.println("Text: " + obj.getText());

        driver.quit();
    }
}
